postgres_insert_block.py

```diff
@@ -0,0 +1,62 @@
+from nio.properties import BoolProperty, StringProperty
+from nio.util.discovery import discoverable
+from .postgres_base_block import PostgresBase
+
+
+@discoverable
+class PostgresInsert(PostgresBase):
+    """A block for inserting incoming signals into a postgres database.
+
+    Properties:
+        table_name(str): name of the table on the database to execute commands
+                         on.
+
+    """
+
+    upsert = BoolProperty(title="Update conflicting keys", default=True)
+    table_name = StringProperty(title="Table name", allow_none=False)
+
+    def __init__(self):
+        super().__init__()
+
+    def process_signals(self, signals):
+        """execute an insert command for all incoming signals"""
+        for signal in signals:
+            self.execute_insert(signal.to_dict())
+
+    def execute_insert(self, data):
+        """execute an insert query for the given data"""
+        self.logger.debug('executing INSERT on data: {}'.format(data))
+        try:
+            self._cur.execute(self._build_insert_query_string(data),
+                              tuple(data.values()))
+        except:
+            self.logger.exception("Could not execute command".format())
+            self._rollback_transactions()
+        else:
+            self._commit_transactions()
```

Comment:
If we're using the commit, does it maybe make more sense to commit after iterating through the whole list of incoming signals? Rather than committing every insert, commit after the whole batch is complete.